certification/internal/engine/engine.go

```diff
@@ -105,12 +108,14 @@ func (c *CraneEngine) ExecuteChecks(ctx context.Context) error {
 			// an error, which requires watching multiple error streams.
 			log.Error("unable to export and flatten container filesystem:", err)
 		}
+		wg.Done()
 	}()
 
 	log.Debug("extracting container filesystem to ", containerFSPath)
 	if err := untar(containerFSPath, r); err != nil {
 		return fmt.Errorf("%w: %s", errors.ErrExtractingTarball, err)
 	}
+	wg.Wait()
```

Comment:
What are doing with this waitgroup? Prepping for more concurrent executions? Fixing a bug?

Reply:
That is correct. Once I enabled tests on this, I immediately got a race condition warning from -race. This fixes it.